app/code/Magento/CatalogSearch/Model/Search/FilterMapper/FilterMapper.php

```diff
@@ -46,14 +46,14 @@ class FilterMapper
      * @param CustomAttributeFilter $customAttributeFilter
      * @param FilterStrategyInterface $filterStrategy
      * @param VisibilityFilter $visibilityFilter
-     * @param StockStatusFilter $stockStatusFilter
+     * @param StockStatusFilterInterface $stockStatusFilter
      */
     public function __construct(
         AliasResolver $aliasResolver,
         CustomAttributeFilter $customAttributeFilter,
         FilterStrategyInterface $filterStrategy,
         VisibilityFilter $visibilityFilter,
-        StockStatusFilter $stockStatusFilter
+        StockStatusFilterInterface $stockStatusFilter
```

Comment:
all the changes to existing code of catalog search - supposed to be backward compatible